packages/x-date-pickers/src/TimeClock/tests/TimeClock.test.tsx

```diff
@@ -20,7 +20,7 @@ describe('<TimeClock />', () => {
     render(<TimeClock value={adapterToUse.date('2019-01-01T04:20:00')} onChange={() => {}} />);
 
     const listbox = screen.getByRole('listbox');
-    expect(listbox).toHaveAccessibleName('Select hours. Selected time is 4:20 AM');
+    expect(listbox).toHaveAccessibleName('Select hours. Selected time is 04:20 AM');
```

Comment:
I think that this is a minor change.
It does not impact the spoken time on VoiceOver or NVDA.